fix: Rate-limiting w/ pauses under 60s, prompt tweaks, button fix

slshults opened the pull request:
Rate-limiting now keeps pauses under 60s to avoid 504s, a few small prompt tweaks, email button properly controlled by billing again.

## Does this work well for both Cloud and self-hosted?
Cloud only

## How did you test this code?
Local, in browser, two concurrent chat sessions in two different browsers until rate-limits exceeded

slshults committed:
fix: Rate-limiting w/ pauses under 60s, prompt tweaks, button by billing fixed

jurajmajerik committed:
feat(experiments): add cumulative exposures chart (#28898)

meikelmosby committed:
fix(cdp): fix wrongful decryption of already decrypted fields (#28915)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

Daesgar committed:
fix: alter the distributed table in coord, instead of the sharded one (#28919)

pauldambra committed:
chore: move two more to playwrt (2 plywrt 2 frious) (#28802)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

Gilbert09 committed:
fix(data-warehouse): Drop `Nones` from incremental value max (#28920)

posthog-bot committed:
chore(deps): Update posthog-js to 1.219.5 (#28921)

Gilbert09 committed:
fix(data-warehouse): Fill nulls when there shouldn't be any nulls (#28923)

Twixes committed:
fix(frontend): Update some api.ts methods for project routing (#28922)

patricio-posthog committed:
revert: "fix: billing limit must be above current usage" (#28857)

andehen committed:
feat(experiments): only show supported math types (#28924)

skoob13 commented:
@slshults I don't quite follow changes here. Can you explain what the problems are?

danielbachhuber committed:
feat(experiments): Wire up support for binomial experiments (#28892)

Co-authored-by: Juraj Majerik <juro.majerik@gmail.com>

jurajmajerik committed:
fix(experiments): cache exposure results for 24h (#28930)

posthog-bot committed:
chore(deps): Update posthog-js to 1.219.6 (#28932)

Gilbert09 committed:
fix(data-warehouse): Cast to the correct tz (#28933)

daibhin committed:
fix: stop guessing issue first seen (#28874)

pauldambra committed:
chore: playwright 11 (#28925)

adamleithp committed:
fix: restore tailwind jit rescan (#28937)

Gilbert09 committed:
fix(data-warehouse): Adjust how we treat NaN/Infinte values with floats and decimals (#28936)

orian committed:
fix: HogQL inefficient timestamp filter (#28912)

HogQL's toDateTime function becomes ClickHouse's toDateTime64 if the input is a literal in ISO format

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

zlwaterfield committed:
feat: customer io email support (#28578)

Co-authored-by: Joshua Snyder <joshua@posthog.com>
Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

pauldambra committed:
chore: more playwright 12 (#28939)

k11kirky committed:
fix: broken links in generation tab (#28907)

pauldambra committed:
chore: lint playwright TS on commit (#28941)

oliverb123 committed:
feat(cymbal): capture some product metrics (#28865)

meikelmosby committed:
feat(cdp): set batch size to meaningful level (#28860)

slshults commented:
> @slshults I don't quite follow changes here. Can you explain what the problems are?

@skoob13  Previously the rate-limiting was only being applied to one of the sessions, while others were able to roll on, which was stretching out the wait times during rate-limiting, and forcing that one rate-limited session to get limited over and over while other sessions could still send messages.  So I added better tracking of tokens and rate-limiting and to make sure rate-limiting gets applied to each session / globally, and reduced the maximum timeout to 40s (gives overhead to keep waits under 60s, at least on my local.

Sriram-bk committed:
fix: Fixed PNG insight exports (#28744)

Twixes committed:
fix(prop-defs): Fix leftover `team_id` prop def lookup in API (#28928)

Twixes committed:
feat(assistant): Make the sidebar Max's home (#28710)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

Gilbert09 committed:
fix(data-warehouse): Normalize primary key (#28940)

aspicer committed:
fix: dashboard resize no mins (#28693)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

Gilbert09 committed:
fix(data-warehouse): Close old django connections in the compaction job (#28910)

Gilbert09 committed:
fix(data-warehouse): Change how we connect with psycopg (#28949)

dmarticus committed:
fix(feature-flags): more cohort name/ID probz (#28905)

Commit:
chore(flags): Add sentry exception capture when error occurs during processing feature flag scheduled changes (#28904)

Gilbert09 committed:
fix(data-warehouse): Even if there are no nulls, update the field type (#28950)

Gilbert09 committed:
fix(data-warehouse): If we cant shape the decimal type, fallback to a default (#28956)

skoob13 commented on ee/support_sidebar_max/sidebar_max_ai.py:
```diff
@@ -65,6 +76,98 @@ def save_to_cache(self, session_id: str, timeout: int = 3600):  # 1 hour default
+    @classmethod
+    def _get_bucket_key(cls, limit_type: RateLimitType) -> str:
+        """Get Redis key for a specific rate limit bucket"""
+        return f"max_rate_limit_bucket_{limit_type}"
```
Let's prefix it like `max_support_*`

slshults replied:
Ah yes, thanks. changed to `support_max_*` in [d7eeabe1b6](https://github.com/PostHog/posthog/commit/d7eeabe1b6)

skoob13 commented on ee/support_sidebar_max/sidebar_max_ai.py:
```diff
@@ -65,6 +76,98 @@ def save_to_cache(self, session_id: str, timeout: int = 3600):  # 1 hour default
+                new_tokens = time_passed * cls.RATE_LIMITS[limit_type]["refill_rate"]
+                tokens = min(tokens + new_tokens, cls.RATE_LIMITS[limit_type]["max"])
+                cache.set(bucket_key, tokens)
+                cache.set(last_update_key, now)
```
The set operation is probably okay for now, but it has race conditions. There is a [better way](https://redis.io/docs/latest/develop/interact/transactions/). It's not a big deal because Anthropic counts tokens for you, so newer requests will always have an updated limit, but edge cases will appear eventually.

skoob13 reviewed (approved):
Let's fix key prefixes and CI. Everything else looks good.

skoob13 reviewed (changes requested):
Sorry, pressed a wrong button😄

slshults committed:
chore: Rename instances of 'max_*' to 'support_max*'

slshults committed:
Merge branch 'master' into support-sidebar-max-rate-limit-fix

slshults committed:
fixed some static typing errors to please mypy

slshults committed:
Merge branch 'master' into support-sidebar-max-rate-limit-fix
